### DISCUSSION TITLE
How to run binary script for the aliased installation?

### DISCUSSION THREAD

**Discussion by kwonoj** [yarnpkg/berry#6564] [Q&A]
Say I have 2 typescript versions installed in package.json

```
{
...,
  "devDependencies": {
    "tsc4": "npm:typescript@4.9.5",
    "typescript": "5.6.2",
  }
}
```

`yarn tsc --version` returns `5.6.2`, which is expected. However, I'm not able to figure out how to run bin for the aliased `tsc4` - something similar to `yarn tsc4 --version` (which doesn't work). Is there a syntax to specify specific dependency to run the binary? So far the only way I was able to find is `yarn dlx -p tsc4@npm:typescript@4.9.5 tsc`, but it seems to invoke temporary installation each time.

**Comment by ShreeChaturvedi** [comment]
There is no `yarn tsc4`. An alias only renames the package. The binary is still `tsc`, so `yarn tsc` is 5.6.2.

PnP (default). Resolve the aliased package and run it:

```bash
yarn node "$(yarn node -p "require.resolve('tsc4/bin/tsc')")" --version
```

Want `yarn tsc4`? Add a script:

```json
"scripts": {
  "tsc4": "node -e \"require('child_process').spawnSync(process.execPath,[require.resolve('tsc4/bin/tsc'),...process.argv.slice(1)],{stdio:'inherit'})\" --"
}
```

node-modules linker:

```bash
node node_modules/tsc4/bin/tsc --version
```

Pitfall: `yarn dlx` reinstalls each time. Use the paths above on the install you already have.
